feat(cgo): Implement graceful shutdown

Fangop opened the pull request:
* The code has been formatted by `gofmt`
* The update of `go.sum` and `go.mod` has been synchronized by `go mod tidy`
* The graceful shutdown of dcurl cgo module will destroy dcurl while receiving `SIGTERM` and `SIGINT` signal as [HORNET](https://github.com/gohornet/hornet/blob/v0.4.1/plugins/gracefulshutdown/plugin.go).
* When destroying dcurl, the logger will log the warning of destroying dcurl.

## Graceful Shutdown
The graceful shutdown implementation requires 2 new go package to be imported.
* The package `syscall` is for the information of system call.
* The package `os` and its sub package `os/signal` is for notifying the signal of system
The concept of the implementation is to start a goroutine when importing dcurl module.
The goroutine will listen to signal of the system.
When receiving `SIGTERM` and `SIGINT`, it'll log the warning and safely destroy dcurl.

## Logger 
The logger is based on the primitive log package in go.
Originally, I use the `go.uber.org/zap`(the logger `iotaledger/hive.go/logger` based on) for the logger.
Consider of the complexity of the synchronization of the loggers, I used primitive log package in go for its simplicity.

Thanks for the reviewing!

junwei0117 commented on cgo/pow/pow.go:
```diff
@@ -36,6 +40,15 @@ func init() {
+	signal.Notify(gracefulShutdown, syscall.SIGTERM, syscall.SIGINT)
+	go func() {
+		<- gracefulShutdown
+		logger.Println("test")
```
Remove testing message.

Fangop replied:
It has been modified to warning message.
Thanks for your  review!

marktwtn commented on cgo/pow/pow.go:
```diff
@@ -18,8 +18,12 @@ char *dcurl_wrapper(char* trytes, int mwm, int threads){
 */
 import "C"
 import (
+	"os"
+	"os/signal"
+	"log"
 	"sync"
 	"unsafe"
+	"syscall"
```
Change to alphabetical order.

Fangop replied:
OK! The request has been done!

Commit:
feat(cgo): Implement graceful shutdown

Gracful shutdown is a feature that allows process safely
shutting down, closing connections, freeing memory space, and so on.
The graceful shutdown implemented in this module is based on
listening to system call.
The shutdown process here will safely destroy dcurl.
